### PR TITLE
[npud] Set request data in Trix backend

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit support to set request data in Trix backend.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

File: runtime/service/npud/backend/trix/TrixBackend.cc [hunk 1]
```diff
@@ -307,8 +307,105 @@ NpuStatus TrixBackend::setRequestData(NpuContext *ctx, RequestID requestId, Inpu
...
+  {
+    return NPU_STATUS_ERROR_INVALID_ARGUMENT;
+  }
+
+  auto &req = _dev->requests.at(requestId);
+  auto miter = std::find(ctx->models.begin(), ctx->models.end(), req->modelId);
+  if (miter == ctx->models.end())
+  {
+    return NPU_STATUS_ERROR_INVALID_MODEL;
+  }
```
Comment (hseok-oh) [inline runtime/service/npud/backend/trix/TrixBackend.cc:328]: Can't we use try-catch from `std::map::at(...)`'s `std::out_of_range` exception?

File: runtime/service/npud/backend/trix/TrixBackend.cc [hunk 2]
```diff
@@ -307,8 +307,105 @@ NpuStatus TrixBackend::setRequestData(NpuContext *ctx, RequestID requestId, Inpu
...
+  auto &minfo = _dev->models.at(req->modelId);
+  if (minfo->meta->input_seg_num != inputBufs->numBuffers ||
+      minfo->meta->output_seg_num != outputBufs->numBuffers)
+  {
+    return NPU_STATUS_ERROR_INVALID_DATA;
+  }
+
+  auto &iter = _dev->requests.at(requestId);
+  auto &inInfos = iter->inInfos;
+  auto &outInfos = iter->outInfos;
```
Comment (hseok-oh) [inline runtime/service/npud/backend/trix/TrixBackend.cc:340]: `_dev->requests.at(requestId)` duplication

```suggestion
  auto &inInfos = req->inInfos;
  auto &outInfos = req->outInfos;
```

File: runtime/service/npud/backend/trix/TrixBackend.cc (hunk 1 above)
Comment (jyoungyun) [reply runtime/service/npud/backend/trix/TrixBackend.cc:328]: Um, before calling `at` function, it already checks whether the `requestId` is included in ctx requests array. Still, should I use `try-catch` phrase?

Comment (jyoungyun) [reply runtime/service/npud/backend/trix/TrixBackend.cc:328]: Oh, do you mention it because of my TODO comment?

Comment (hseok-oh) [reply runtime/service/npud/backend/trix/TrixBackend.cc:328]: >Um, before calling `at` function, it already checks whether the `requestId` is included in ctx requests array. Still, should I use `try-catch` phrase?

My comment means that we can remove `std::find(...)` checking if we use exception.

Comment (jyoungyun) [reply runtime/service/npud/backend/trix/TrixBackend.cc:328]: @hseok-oh 
Um, I can't find a way to declare the reference variable of the unique_ptr item outside the `try-catch` block. I don't want to use `get()` function of unique_ptr... Could you suggest an elegant way to do this?

```
  TrixRequestInfo *req;
  try
  {
    req = _dev->requests.at(requestId).get();
  }
  catch(const std::exception &e)
  {
    return NPU_STATUS_ERROR_INVALID_ARGUMENT;
  }
```

Comment (hseok-oh) [reply runtime/service/npud/backend/trix/TrixBackend.cc:328]: I also don't have any idea for elegant implementation. :)

**Comment by jyoungyun** [comment]
PTAL

**Comment by jyoungyun** [comment]
PTAL. Thanks in advance. :)

**Review by hseok-oh — APPROVED** [review]

**Pull request merged**